### DISCUSSION TITLE
Effect of r_max on performance and accuracy.

### DISCUSSION THREAD

**Discussion by JSLJ23** [ACEsuit/mace#464] [Q&A] [answered]
Hi MACE developers,

Just wanted to better understand how the `r_max` parameter affects the underlying MACE model in terms of performance and accuracy? i.e. Does increasing the r_max beyond the default 5.0A slow down training/inference speed and increase memory consumption? Does a larger r_max lead to better errors in the energy and forces during training?

Also a brief explanation on why `r_max` might or might not affect the above would be greatly appreciated.

**Comment by ilyes319** [comment, marked as answer]
Hi @JSLJ23,

Larger r_max will slow down the model, as it will increase the number of edges per atom. 

The right r_max to use really depends on the system. In general, I would never go beyond 10A, and a safe bet is to use 6A or 5A. Some element have very large covalent radii and therefore require larger r_max. Some systems are more short range than other and could get away with shorter r_max (like 4A). Usually, the plot of accuracy vs r_max is a parabola, and increasing r_max beyond a certain point can make the fit worse.

A good heuristic for choosing the r_max is to look at the average number of neighbors. This number should be between 10 and 60 for best performance, and should not typically go above 100.

If you want to scan the r_max for your system, I would try 4.5A, 5.0A, 6.0A and 7A.

**Reply by JSLJ23** [reply]
Thank you for such a detailed response and also some guidelines on how to go about trying out different `r_max` values.

To give a bit more background context behind this question, I am trying to train a version of MACE that can predict interaction energies between key amino acid residues in a protein pocket and a small molecule ligand. Some of these residues might have atoms that span more than 10A away from the **furthest** ligand atom and my thought process was along the lines of if those amino acid atoms are not able to "see" and "interact" with the small molecule atoms, there might be insufficient information to learn the embedding behind the interaction energy (i.e. `MACE_energy_of_amino_acid_and_ligand` - `MACE_energy_of_amino_acid` - `MACE_energy_of_ligand`). Do you guys happen to have any thoughts on this or is a `r_max` cutoff less than the longest inter-atomic distance sufficient?

**Reply by gabor1sg** [reply]
are you trying to do this on a fixed geometry that you perhaps download from the PDB and the ligand is generated by some other docking program? or are you trying to create a force field that can be used to do MD on this system?

**Reply by JSLJ23** [reply]
It is from a fixed geometry but from MD frames of a trajectory where the training data is the ab initio interaction energy (i.e. `PSI4_DFT_energy_of_amino_acid_and_ligand` - `PSI4_DFT_energy_of_amino_acid` - `PSI4_DFT_energy_of_ligand`) from something like [this](https://docs.eyesopen.com/floe/modules/psi4-orion/docs/source/floes/Psi4_QM_Interaction_Energy.html).

**Reply by gabor1** [reply]
I see. so you are making really sort of a force field, but its target is not the total energy, but the interaction energy between the ligand and the protein. quite sensible thing to do. so in this case I would not worry about a cutoff of 5 or 6. the point is that the interaction energy has a much tighter locality than the _free energy_ of binding. for the latter, lots of molecular motions are relevant, and things further away also come in play through the boltzmann distribution. but for the pointwise energy fitting, a mace with 2 layers and a layer-cutoff of 5 or 6 should be sufficient.

**Reply by gabor1** [reply]
how do you plan to use this model that you made? I can imagine that you can run classical empirical force field MD, sample it, and compute the average binding energy using your model on these snapshots. alternatively, you could try to make a combined model (e.g. classical empirical force field for protein and ligand separately + your new model) and use this to run the actual MD
